### ISSUE TITLE
NullReferenceException in MvxAndroidBindingContextHelpers.Current

### ISSUE THREAD

**Issue by amulware** [toggl/mobileapp#2923]
## 💥 Crash Report
By initialising app center earlier, we are getting a lot more crash reports which hopefully will account for the so far unexplained crashes (see #2833).

### 👥 Impact
- about 1600 crashes, over 500 users, for 84% of all crashes in 1.3
- all sorts of devices and OS versions

### 🛰 Link
https://appcenter.ms/orgs/Toggl/apps/Toggl-Android/crashes/groups/96ea405eecd1afb1ffe35f417763a4f48b2a2ba5?appVersion=1.3
https://appcenter.ms/orgs/Toggl/apps/Toggl-Android/crashes/groups/8153f92a96249266df0bb5c054e7b125e7947458?appVersion=1.3

### 📝  Exception & Stack Trace
`System.NullReferenceException: Object reference not set to an instance of an object`
```cs
Mvx.Resolve[TService] ()
MvxAndroidBindingContextHelpers.Current[T] ()
MvxAndroidBindingContextHelpers.Current ()
MvxLayoutInflater.Inflate (System.Int32 resource, Android.Views.ViewGroup root, System.Boolean attachToRoot)
LayoutInflater.n_Inflate_ILandroid_view_ViewGroup_Z (System.IntPtr jnienv, System.IntPtr native__this, System.Int32 resource, System.IntPtr native_root, System.Boolean attachToRoot)
```

**Comment by amulware** [comment]
This is still happening in 1.3.1 :(

**Comment by daividssilverio** [comment]
I actually got this bug also while doing some testing. It happens sometimes when you tap the fab to start a new time entry. (specially when there's no TEs at all)

**Comment by simonrozsival** [comment]
This is the best way to make sure that the data source is properly created and registered as a service currently. I would suggest adding a simple method called something like `InitializeDataSourceIfLoggedIn()` to the data source which would do just that, but it would not seem like a WTF unnecessary bunch of lines in the `App.cs` file (even though you could describe everything in a comment, well you definitely should in this case).

**Comment by dshr** [comment]
`var loginManager = Mvx.Resolve<ILoginManager>();` doesn't really resolve it — I'm still getting the same error every once in a while. I've tried adding some `Console.WriteLine`s around that line, and the crashes happen before that.

**Comment by daividssilverio** [comment]
Sorry I forgot to say @dshr, those lines can be added on Giskard's `Setup:InitializeApp` right before `base.InitializeApp(pluginManager, app);`

**Comment by dshr** [comment]
@daividssilverio tested a bit further today — managed to get it not to break even with the `ChangeStatusBarColor` calls, by adding the dreaded `var MvxSetup = MvxAndroidSetupSingleton.EnsureSingletonAvailable(ApplicationContext); MvxSetup.EnsureInitialized();` right before it in all activities that can be accessed on launch. 

Funnily enough, moving that code block to a common place, like `Setup.cs` causes it to throw a `EnsureInitialized has already been called so now waiting for completion`. Pretty bizarre.

IMO, having less crashes > having clean code, so I suggest we test this again on Monday and release it as 1.8.2 if all goes well.

**Comment by dshr** [comment]
Fixed in #2999
